src/frontend/planner_test/tests/testdata/dynamic_filter.yaml

```diff
@@ -0,0 +1,119 @@
+# This file is automatically generated. See `src/frontend/planner_test/README.md` for more information.
+- id: create_tables
+  sql: |
+    create table t1 (v1 int);
+    create table t2 (v2 int);
+- name: With Simple Agg on inner side
+  before:
+  - create_tables
+  sql: |
+    with max_v2 as (select max(v2) max from t2) select v1 from t1, max_v2 where v1 > max;
+  optimized_logical_plan: |
+    LogicalJoin { type: Inner, on: (t1.v1 > max(t2.v2)), output: [t1.v1] }
+    ├─LogicalScan { table: t1, columns: [t1.v1] }
+    └─LogicalAgg { aggs: [max(t2.v2)] }
+      └─LogicalScan { table: t2, columns: [t2.v2] }
+  stream_plan: |
+    StreamMaterialize { columns: [v1, t1._row_id(hidden)], pk_columns: [t1._row_id] }
+    └─StreamDynamicFilter { predicate: (t1.v1 > max(max(t2.v2))), output: [t1.v1, t1._row_id] }
+      ├─StreamTableScan { table: t1, columns: [t1.v1, t1._row_id], pk: [t1._row_id], dist: UpstreamHashShard(t1._row_id) }
+      └─StreamExchange { dist: Broadcast }
+        └─StreamProject { exprs: [max(max(t2.v2))] }
+          └─StreamGlobalSimpleAgg { aggs: [sum(count), max(max(t2.v2))] }
+            └─StreamExchange { dist: Single }
+              └─StreamHashAgg { group_key: [Vnode(t2._row_id)], aggs: [count, max(t2.v2)] }
+                └─StreamProject { exprs: [t2.v2, t2._row_id, Vnode(t2._row_id)] }
+                  └─StreamTableScan { table: t2, columns: [t2.v2, t2._row_id], pk: [t2._row_id], dist: UpstreamHashShard(t2._row_id) }
+- name: |
+    With Top-1 on inner side
+    TODO: currently not possible due to https://github.com/risingwavelabs/risingwave/issues/5764
+  before:
+  - create_tables
+  sql: |
+    with max_v2 as (select v2 max from t2 order by v2 desc limit 1) select v1 from t1, max_v2 where v1 > max;
```

Comment:
Interesting.

Reply:
Some off-topic idea. Maybe we can support more range conditions than simple `>`, `>=`, `<`, `<=`.
e.g. `between`
```sql
 with max_v2 as (select max(v2) max from t2) select v1 from t1, max_v2 where v1 between max - 100 and max + 100;
```

Reply:
Exactly, we can support arbitrary range in the future.